B/Birch/build_tarballs.jl

```diff
@@ -0,0 +1,49 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Birch"
+version = v"1.0.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/lawmurray/Birch.git", "a533990b59b04ef4cd20f8d4fe91c4a220df1cfa"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+# install the driver
+cd ${WORKSPACE}/srcdir/Birch/driver/
+autoreconf -vi
+CPPFLAGS="-I${prefix}/include" ./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+
+# install license
+install_license LICENSE
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms())
+
+# GLOB_NOMAGIC is not supported by musl
+filter!(p -> libc(p) != "musl", platforms)
+
+# Native support for Windows is not yet provided
+filter!(!Sys.iswindows, platforms)
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("birch", :birch),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+    Dependency("boost_jll"),
+    Dependency("LibYAML_jll"),
```

Comment:
@lawmurray I'm unsure about the runtime dependencies after splitting it into multiple packages, even though you provided some explanations in the previous discussion [here](https://github.com/JuliaPackaging/Yggdrasil/pull/1924#discussion_r507227011). I did not include Eigen as a dependency here since it is not required for building `birch` - but is it needed at runtime? Or only by e.g. the standard library? And are these dependencies listed above needed at runtime?

Similary, should LibBirch state any runtime dependencies (currently I only added Eigen as build dependency in https://github.com/JuliaPackaging/Yggdrasil/pull/2155)?

Reply:
And similarly, are the dependencies of the standard library in https://github.com/JuliaPackaging/Yggdrasil/pull/2157 correct?

Reply:
Yes, that's right. The dependencies for all packages are essentially the same. Exceptions:

1. Eigen is only needed by LibBirch and Birch packages, not the compiler.
2. The Boost Filesystem and Boost System libraries are not needed by LibBirch.

As in my other comments, just translate what's in the Homebrew and Arch Linux packages I'd say.

Reply:
OK, I think then my intuition was (somewhat) correct here :slightly_smiling_face:  I'll also check the PKGBUILD files :+1: